src/poetry/core/version/markers.py

```diff
@@ -244,8 +247,25 @@ def validate(self, environment: dict[str, Any] | None) -> bool:
             return True
 
         # The type of constraint returned by the parser matches our constraint: either
-        # both are BaseConstraint or both are VersionConstraint.  But it's hard for mypy
-        # to know that.
+        # both are BaseConstraint or both are VersionConstraint. But it's hard for mypy
```

Comment:
you have increased the distance between this comment and the code that it is talking about; and increased the verbiage to compensate

surely better just to leave the comment and the code next to each other?!

Reply:
You are right. In an earlier version, I used `allows()` and needed two more ignores before I realized I can just use `in` / `not in` since I have to make a distinction anyway.